bot/exts/moderation/incidents.py

```diff
@@ -402,10 +486,147 @@ async def on_raw_reaction_add(self, payload: discord.RawReactionActionEvent) ->
+        webhook message for that particular link from the channel.
+
+        If the message edit (`msg_after`) is a incident then run it through `extract_message_links`
+        to get all the message link embeds (embeds which contain information about that particular
+        link), this message link embeds are then sent into the channel.
+
+        The edited message is also passed into `add_signals` if it is a incident message.
+        """
+        try:
+            channel = self.bot.get_channel(int(payload.data["channel_id"]))
+            msg_after = await channel.fetch_message(payload.message_id)
+        except discord.NotFound:  # Was deleted before we got the event
+            await self.delete_msg_link_embed(payload.message_id)
+            return
+
+        if is_incident(msg_after):
+            webhook_embed_list = await self.extract_message_links(msg_after)
+            webhook_msg_id = await self.message_link_embeds_cache.get(payload.message_id)
+
+            if not webhook_embed_list:
+                await self.delete_msg_link_embed(msg_after.id)
+                return
+
+            if webhook_msg_id:
+                await self.incidents_webhook.edit_message(
+                    message_id=webhook_msg_id,
+                    embeds=[embed for embed in webhook_embed_list if embed is not None],
+                )
+                return
+
+            await self.send_message_link_embeds(webhook_embed_list, msg_after, self.incidents_webhook)
+
+    @Cog.listener()
+    async def on_raw_message_delete(self, payload: discord.RawMessageDeleteEvent) -> None:
+        """
+        Delete message link embeds for `payload.message_id`.
+
+        Search through the cache for message, if found delete it from cache and channel.
+        """
+        await self.delete_msg_link_embed(payload.message_id)
+
+    async def extract_message_links(self, message: discord.Message) -> t.Optional[t.List[discord.Embed]]:
+        """
+        Checks if there's any message links in the text content.
+
+        Then passes the the message_link into `make_message_link_embed` to format a
+        embed for it containing information about the link.
+
+        As discord only allows a max of 10 embeds in a single webhook, just send the
+        first 10 embeds and don't care about the rest.
+
+        If no links are found for the message, it logs a trace statement.
+        """
+        message_links = DISCORD_MESSAGE_LINK_RE.findall(str(message.content))
+
+        if message_links:
+            embeds = []
+            for message_link in message_links[:10]:
+                ctx = await self.bot.get_context(message)
+                embed = await make_message_link_embed(ctx, message_link[0])
+                if embed:
+                    embeds.append(embed)
+
+            return embeds
+
+        log.trace(
+            f"Skipping discord message link detection on {message.id}: message doesn't qualify."
+        )
```

Comment:
This seems inaccurate. Detection wasn't skipped, it just didn't detect anything. :smile:

Reply:
Ah yes 😅